_posts/2020-03-19-our-rails-upgrade-process.markdown

```diff
@@ -0,0 +1,83 @@
+
+A good examples is the deprecation warnings caused by the changes in `ActiveRecord::Dirty` in the Rails 5.2 version. We wrote [this article](https://www.fastruby.io/blog/rails/upgrades/active-record-5-1-api-changes.html) while ago showing how to address them.
+
+## Step 3: Add dual boot for the Rails version
+
+To help us switch between the current Rails version and the new one, we usually create a dual boot mechanism. The fastest way is to install the handful gem [next_rails](https://github.com/fastruby/next_rails). Please, [visit this article](https://www.fastruby.io/blog/upgrade-rails/dual-boot/dual-boot-with-rails-6-0-beta.html) where we showed how you can install and setup the gem in your local environment and your CI server.
+
+
+## Step 4: Assessing whether we can upgrade a dependency or not
+
+After shipping all deprecation warnings fixes to the master branch and setup the dual boot, it's time for us to handle with the project dependencies.
+
+Sometimes dependencies are backwards compatible with the current version of Rails. They will have a code like:
+
+```
+if Rails::VERSION >= 5.1
+  # does X
+else
+  # does Y
+end
+```
+
+If that is the case, then you might be able to just upgrade the dependency using `bundle update`.
+
+If the dependency does not have backwards compatible code, then, using the next_rails gem, you can add something like this to your application's Gemfile:
+
+```
+if next?
+  gem "dep", "~> 1.2.3"
+else
+  gem "dep", "~> 0.9"
+end
+```
+
+## Step 5: Create the rails upgrade branch and submit a Pull request
+
+After adjust all the dependencies in the application for the both Rails versions, we create the `rails-next-version` branch with all the dual boot code and open a PR that will target to master. The idea is that from now one, every PR that we create and we can't merged to master directly, will target to this `rails-next-version` branch. As you can imagine, this big PR won't be merged until all necessary changes to the upgrade are done.
+
+This help us to deliver small testable changes during the upgrade and keep a stable branch that we can use as base.
+
+## Step 6: Create new branches for each update that is needed
+
+For every story that we're working on, we create a new pull request. If the change is not backwards compatible the PR will target to the `rails-next-version` branch, if it is, it will target to the `master` branch.
+
+At this point if we run the test suite using the Rails 5.2 changes, it's probably that a bunch of tests will fail. It's time to address them and open a pull request to each one or for each file or feature, depending of the complexity of the changes.
+
+## Last Step:  Merge the rails upgrade branch
+
```

Comment:
I think we need a step before this one which talks about doing quality assurance in a QA environment. There are usually visual issues that only come up when you do “real integration testing”